### ISSUE TITLE
Printing in firefox fails

### ISSUE THREAD

**Issue by mduveborg** [DoersGuild/jQuery.print#19]
Printing in firefox 37 does not work. 

Adding this code in the try-catch at line 132 makes it work again:

 var printed = w.document.execCommand('print', false, null);
                        if (!printed) {
                            w.print();
                        }

**Comment by sathvikp** [comment]
What version of the plugin are you using?
On the latest (1.3.2), we have that on [line 31](https://github.com/DoersGuild/jQuery.print/blob/1a5fae505e4bc72438f46c0865a3bcc1782be04e/jQuery.print.js#L31) as

```js
               try {
                    // Fix for IE11 - printng the whole page instead of the iframe content
                    if (!frameWindow.document.execCommand('print', false, null)) {
                        // document.execCommand returns false if it failed -http://stackoverflow.com/a/21336448/937891
                        frameWindow.print();
                    }
                } catch (e) {
                    frameWindow.print();
                }
```

**Comment by ssuess** [comment]
Thought I was losing my mind, but yeah I can't print reliably in Firefox anymore (tried versions 37.0.1 and 37.0.2 and 38 beta). It will randomly print when clicked sometimes, other times not.

**Comment by ssuess** [comment]
I also am not sure where you are inserting that code and in which version, could you please explain? I really need to get this working in Firefox, because it is the only browser that supports printing table headers on every page.

**Comment by ssuess** [comment]
I take it back, I reverted to an earlier Firefox and it isn't working there either. I am not sure at all what could have changed, but I get no error message of any significance. It simply won't launch the print dialog. Chrome and Safari will launch the print dialog every time my button is pressed, but of course they don't support table header printing so I can't use them. Any ideas?

**Comment by ssuess** [comment]
I figured out why this was failing, and why I didn't see it in my earlier testing: This is entirely due to the distance from the server/speed of the network connection to the server. When I tried my code on localhost, it would always work. When I tried connecting to that webserver remotely, it would only sometimes work. It is entirely due to that! Now, the question is, what can we do about it? Can we introduce some pause in the program to give it time to collect all the elements, create the iframe, and call the print dialog?

**Comment by ssuess** [comment]
Nevermind, I found it. I just changed the value in your setTimeout function to 1500 (from 250) and voila!

**Comment by sathvikp** [comment]
So, to understand this better, you're loading external CSS and so when you print, it shows you unstyled content? @ssuess

**Comment by ssuess** [comment]
no, before changing the timeout value, it would either work or not work. When it worked, everything would print just fine. When it didn't, the print dialog would never show at all, even though there was a little activity of some sort when clicking my print button. Changing the timeout value allows the page enough time to copy all the elements to the iframe for printing. Now the print dialog will come up 100% of the time, albeit with a delay of a second or so, which is fine for my purpose. I am not entirely sure this is the same problem that the person who opened this issue was having, but possibly it was. I would be curious if changing the setTimeout value for them fixed the problem as well.

**Comment by sathvikp** [comment]
That is quite odd. 
Do you have any script tags in the element you've selected to print?
Because scripts can block the rest of the content from loading while they download & execute.
@ssuess

**Comment by ssuess** [comment]
I did have a script tag I guess (in the sense that I had previously embedded your code in the very div I was printing. So thinking that might have been the cause as you suggest, I separated it out and made sure I had nothing inside my print div other than html, although there are some js calls (onmouseover, onclick) on some of the elements inside this div, not sure if that makes any difference. In any event, this made no difference, I still had to introduce the extra second or so into setTimeout or it fails the same way as before.
